packages/helpermodules/timecheck.py

```diff
@@ -236,6 +236,13 @@ def create_unix_timestamp_current_full_hour() -> int:
     return int(datetime.datetime.strptime(full_hour, "%m/%d/%Y, %H").timestamp())
 
 
+def create_unix_timestamp_current_quarter_hour() -> int:
+    def round_to_quarter_hour(current_time: float, quarter_hour: int = 900) -> float:
+        log.debug(f"current time: {current_time} => modified: {current_time - (current_time % quarter_hour)}")
+        return current_time - (current_time % quarter_hour)
+    return int(round_to_quarter_hour(datetime.datetime.today().timestamp()))
+
+
```

Comment:
@LKuemmel was hältst Du davon?
```suggestion

def __calculate_price_timeslot_length(self, prices: dict[str, int]) -> int:
    first_timestamps = list(prices.keys())[:2]
    return int(first_timestamps[1]) - int(first_timestamps[0])


def select_unix_timestamp_current_price_slot(prices: dict[str, float]) -> int:
    now = int(datetime.datetime.today().timestamp())
    price_timeslot_seconds = self.__calculate_price_timeslot_length(prices)
    return next(iter( [int(timestamp) for timestamp in list(prices.keys()) if (
        int(timestamp) >= now and int(timestamp) <= now -  price_timeslot_seconds
    )]))
```